### DISCUSSION TITLE
Noob question: How can I make a form editor?

### DISCUSSION THREAD

**Discussion by SuperPauly** [eclipse-theia/theia#10342] [Q&A] [answered]
I have seen the sprotty GLSP demo but it seems a lot to setup and I don't really want diagrams.

I just a form with a few fields in like name, phone, age and the ability to add a new field and to be able to drag it around, zoom in and out. 

How would I go about doing this? 

Would this definatly need GLSP/Sprotty or could I just build it with a Extension/Plugin?

I've had a look at the Theia TypeDoc API but not much in depth explaining and also the examples link is
broken https://eclipse-theia.github.io/theia/docs/next/index.html 

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @SuperPauly, thank you for your question.

The TypeDoc is often not really helpful in that regard. We also have a [`theia-ide.org`](https://theia-ide.org/docs/) domain with lots of helpful documentation to get started with Theia. However the documentation on how to add custom widgets is currently in a proposal state (see https://github.com/eclipse-theia/theia-website/pull/214). You can see a preview right [here](https://deploy-preview-214--romantic-mcclintock-8d974f.netlify.app/docs/widgets/). That should explain how to build a custom widget by just using a Theia extension. In there, you are free to render any HTML/React content you like.

Alternatively you can also just create a vscode extension that contributes a [webview](https://code.visualstudio.com/api/extension-guides/webview) and consume that in Theia.

**Reply by SuperPauly** [reply]
Thanks @msujew  :)
